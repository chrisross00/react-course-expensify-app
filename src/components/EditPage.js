import React from 'react';
import { connect } from 'react-redux';
import ExpenseForm from './ExpenseForm';
import { startEditExpense, removeExpense, startRemoveExpense } from '../actions/expenses';

// Refactor EditExpensePage to be a class based component
// Setup mapDispatchToProps editExpense and removeExpense
//

export class EditPage extends React.Component {
  onSubmit = (expense) => {
    this.props.startEditExpense(this.props.expense.id, expense);
    this.props.history.push('/');
  };
  onClick = () => {
    this.props.startRemoveExpense({ id: this.props.expense.id });
    this.props.history.push('/');
  };
  render() {
    return (
      <div>
        <div className="page-header">
          <div className="content-container">
            <h2 className="page-header__title">
              Edit Expense
            </h2>
          </div>
        </div>
        <div className="content-container">
          <ExpenseForm
            className="form"
            expense={this.props.expense}
            onSubmit={this.onSubmit}
          />
          <button
            className="button button--secondary"
            onClick={this.onClick}>Remove Expense
          </button>
        </div>

      </div>
    );
  }
}

const mapDispatchToProps = (dispatch, props) => ({
  startEditExpense: (id, expense) => dispatch(startEditExpense(id, expense)),
  startRemoveExpense: (expense) => dispatch(startRemoveExpense(expense))
});

const mapStateToProps = (state, props) => {
  return {
    expense: state.expenses.find((expense) => expense.id === props.match.params.id)
  };
}


export default connect(mapStateToProps, mapDispatchToProps)(EditPage);
