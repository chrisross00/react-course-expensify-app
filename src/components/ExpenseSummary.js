import React from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import expenseFilter from '../selectors/expenses';
import expensesTotal from '../selectors/expenses-total';
import numeral from 'numeral';

export const ExpenseSummary = ({ vizExpenses, expenseTotal }) => {
  const expenseWord = (vizExpenses.length === 1) ? "expense" : "expenses"
  const expenseAmount = numeral(expenseTotal / 100).format('$0,0.00');
  return (
    <div className="page-header">
      <div className="content-container">
        <h2 className="page-header__title">
          Viewing <span>{vizExpenses.length}</span> {expenseWord} totaling <span>{expenseAmount}</span>
        </h2>
        <div className="page-header__actions">
          <Link className="button" to="/create">Add Expense</Link>
        </div>
      </div>
    </div>
  );
};

const mapStateToProps = (state) => {
  return {
    vizExpenses: expenseFilter(state.expenses, state.filters),
    expenseTotal: expensesTotal(expenseFilter(state.expenses, state.filters))
  };
}

export default connect(mapStateToProps)(ExpenseSummary);
