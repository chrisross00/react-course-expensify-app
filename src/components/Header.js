import React from 'react';
import ReactDom from 'react-dom';
import { NavLink } from 'react-router-dom';

const Header = () => (
  <header>
    <h1>Expensify</h1>
    <NavLink to="/" activeClassName="is-active" exact={true}>Home</NavLink>
    <NavLink to="/create" activeClassName="is-active">Add an Expense</NavLink>
    <NavLink to="/help" activeClassName="is-active">Help and FAQs</NavLink>
  </header>
);

export default Header;
