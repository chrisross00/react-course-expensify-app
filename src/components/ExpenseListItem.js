import React from 'react';
import { connect } from 'react-redux';
import { removeExpense } from '../actions/expenses';
import EditPage from '../components/EditPage';
import { Link } from 'react-router-dom';

export default ({ dispatch, description, amount, createdAt, id }) => (
  <div>
    <Link to={`/edit/${id}`}>
      <h3>{description}</h3>
    </Link>
    <p>{amount} - {createdAt}</p>
  </div >
);
