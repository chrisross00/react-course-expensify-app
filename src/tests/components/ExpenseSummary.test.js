import React from 'react';
import { shallow } from 'enzyme';
import { ExpenseSummary } from '../../components/ExpenseSummary';
import moment from 'moment';
import expensesTotal from '../../selectors/expenses-total';


const expenses = [{
  id: '1',
  description: 'Gum',
  note: '',
  amount: 9000,
  createdAt: moment(0).subtract(2, 'months')
}, {
  id: '2',
  description: 'Rent',
  note: '',
  amount: 443,
  createdAt: moment(0).add(2, 'days')
}];

test('Should see no expenses', () => {
  const wrapper = shallow(
    <ExpenseSummary
      vizExpenses={[]}
    />
  );
  expect(wrapper).toMatchSnapshot();
});

test('Should see two expenses totaling $94.43', () => {
  const expense = expenses;

  const wrapper = shallow(
    <ExpenseSummary
      vizExpenses={expense}
      expenseTotal={expensesTotal(expense)}
    />
  );
  expect(wrapper).toMatchSnapshot();
});

test('Should see one expense totaling $90.00', () => {
  const expense = [expenses[0]];
  const wrapper = shallow(
    <ExpenseSummary
      vizExpenses={expense}
      expenseTotal={expensesTotal(expense)}
    />
  );
  expect(wrapper).toMatchSnapshot();
});
