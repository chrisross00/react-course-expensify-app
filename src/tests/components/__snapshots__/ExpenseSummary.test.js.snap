// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should see no expenses 1`] = `
<div
  className="page-header"
>
  <div
    className="content-container"
  >
    <h2
      className="page-header__title"
    >
      Viewing 
      <span>
        0
      </span>
       
      expenses
       totaling 
      <span>
        $0.00
      </span>
    </h2>
    <div
      className="page-header__actions"
    >
      <Link
        className="button"
        replace={false}
        to="/create"
      >
        Add Expense
      </Link>
    </div>
  </div>
</div>
`;

exports[`Should see one expense totaling $90.00 1`] = `
<div
  className="page-header"
>
  <div
    className="content-container"
  >
    <h2
      className="page-header__title"
    >
      Viewing 
      <span>
        1
      </span>
       
      expense
       totaling 
      <span>
        $90.00
      </span>
    </h2>
    <div
      className="page-header__actions"
    >
      <Link
        className="button"
        replace={false}
        to="/create"
      >
        Add Expense
      </Link>
    </div>
  </div>
</div>
`;

exports[`Should see two expenses totaling $94.43 1`] = `
<div
  className="page-header"
>
  <div
    className="content-container"
  >
    <h2
      className="page-header__title"
    >
      Viewing 
      <span>
        2
      </span>
       
      expenses
       totaling 
      <span>
        $94.43
      </span>
    </h2>
    <div
      className="page-header__actions"
    >
      <Link
        className="button"
        replace={false}
        to="/create"
      >
        Add Expense
      </Link>
    </div>
  </div>
</div>
`;
